packages/dds/sequence/src/sharedStringWithSequentialId.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Lazy } from "@fluidframework/common-utils";
+import { IChannelAttributes, IFluidDataStoreRuntime } from "@fluidframework/datastore-definitions";
+import { Marker, MergeTreeDeltaType, MergeTreeMaintenanceType,
+     reservedMarkerIdKey, SortedSegmentSet } from "@fluidframework/merge-tree";
+import { createIdAfterMin } from "./generateSequentialId";
+import { SequenceMaintenanceEvent } from "./sequenceDeltaEvent";
+import { SharedString } from "./sharedString";
+
+export function sharedStringWithSequentialIdMixin(Base: typeof SharedString = SharedString): typeof SharedString {
+    return class SequenceWithSequentialId extends Base {
+        private readonly sortedMarkers: Lazy<SortedSegmentSet>;
+        constructor(document: IFluidDataStoreRuntime, public id: string, attributes: IChannelAttributes) {
+            super(document, id, attributes);
+            this.sortedMarkers = new Lazy<SortedSegmentSet>(() => {
+                const set = new SortedSegmentSet();
+                this.walkSegments((segment) => {
+                    if (Marker.is(segment) && segment.getId() !== undefined) {
+                        // If it has an id then we have acked it
+                        set.addOrUpdate(segment);
+                    }
+                    return true;
+                });
+                return set;
+            });
+            this.on("maintenance", this.applySequentialId);
```

Comment:
don't we need to do the same thing on sequence delta for remote inserts/deletes? ack will only happen for local

Reply:
Yes just added this! Thanks for reminding!